### PR TITLE
Upgrade to 2.30, drop Python 2.7

### LINKED ISSUE #15: Drop Python 2 support
Currently the build is failing due to Python 2 incompatibility. I can patch it, but my plan was to make the current release (2.30) the last one supporting Python 2. But it might be easier just to drop it, since this has already happened with matplotlib, scipy, numpy, etc.

### PR DESCRIPTION
**Pull request by stuarteberg**
These are @count0's commits to upgrade to graph-tool 2.30. This PR also drops support for Python 2.7, as discussed in #15.

(I ran `conda smithy rerender` on my local machine, so there's no need to ask the bot to do it below.  Python 2.7 support was dropped in the rerendering process, since conda-forge has [officially deprecated][1] Python 2.7 now.)

Fixes #15

[1]: https://gitter.im/conda-forge/conda-forge.github.io?at=5e73e282f5b9723ad9ae09f0

---

Checklist
* [x] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)

### CODE REVIEW

**Comment by stuarteberg** [comment]
Hmm, looks like the linter was [recently changed to be more strict about license names][1], hence the little complaint above.

Looks like these are [the current choices][2]:

- GPL-3.0-only
- GPL-3.0-or-later

Which of those is correct?

[1]: https://github.com/conda-forge/conda-smithy/issues/1265
[2]: https://github.com/conda-forge/conda-smithy/blob/38581061c88c6699581089fb1c94eacb07679dad/conda_smithy/licenses.txt#L168-L169

**Comment by count0** [comment]
GPL-3.0-or-later is the correct one.

**Comment by stuarteberg** [comment]
OK, I updated the license identifier and the linter is happy.

Just to summarize:

Assuming the build passes, this PR can be merged (by any of us), ~5 hours from now.  After that, the CI for the master branch will begin, and THAT build will be automatically uploaded to the `conda-forge` channel.  Then there's a slight delay before Anaconda's CDN makes those packages visible to all conda users.

Basically, graph-tool 2.30 will be available to all** conda users sometime tomorrow, hopefully. :-)

** Except for Python-2 users, of course.

**Comment by stuarteberg** [comment]
BTW, I know we already decided to drop support for Python 2, but if you want some sense of how many users might be affected, one place to look is our download counts on anaconda.org, for files with `py27` in the name:

https://anaconda.org/conda-forge/graph-tool/files

Of course, it's difficult to know how many of those are CI systems, or how many users are "double-counted" because they downloaded more than once, etc.

**Comment by count0** [comment]
Ok great, thanks.

I'm not too worried about Python 2.  People who choose to go down that path are comfortable with getting stuck in time. And I'm happy with diminishing my testing and maintenance surface.

**Comment by stuarteberg** [comment]
Passed on linux 3.6/3.7/3.8, and on mac 3.6/3.8, but the CI container for 3.7 failed to start for some reason.  I'm sure it would have passed, so merging now.  Hopefully the CI for the master branch succeeds completely.
